src/java/fr/paris/lutece/plugins/extend/service/extender/ResourceExtenderService.java

```diff
@@ -1,565 +1,588 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * License 1.0
+ */
+package fr.paris.lutece.plugins.extend.service.extender;
+
+import fr.paris.lutece.plugins.extend.business.extender.IResourceExtenderDAO;
+import fr.paris.lutece.plugins.extend.business.extender.ResourceExtenderDTO;
+import fr.paris.lutece.plugins.extend.business.extender.ResourceExtenderDTOFilter;
+import fr.paris.lutece.plugins.extend.service.ExtendPlugin;
+import fr.paris.lutece.plugins.extend.service.ExtendableResourceResourceIdService;
+import fr.paris.lutece.plugins.extend.service.IExtendableResourceManager;
+import fr.paris.lutece.plugins.extend.service.type.IExtendableResourceTypeService;
+import fr.paris.lutece.portal.business.user.AdminUser;
+import fr.paris.lutece.portal.service.plugin.Plugin;
+import fr.paris.lutece.portal.service.rbac.RBACService;
+import fr.paris.lutece.portal.service.resource.IExtendableResource;
+import fr.paris.lutece.portal.service.util.AppLogService;
+import fr.paris.lutece.util.ReferenceList;
+
+import org.apache.commons.lang3.StringUtils;
+import jakarta.transaction.Transactional;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.inject.Named;
+import jakarta.inject.Inject;
+import jakarta.enterprise.context.ApplicationScoped;
+
+import fr.paris.lutece.portal.service.cache.Lutece107Cache;
+import fr.paris.lutece.portal.service.cache.LuteceCache;
+
+/**
+ *
+ * ResourceExtenderService
+ *
+ */
+@ApplicationScoped
+@Named( "extend.resourceExtenderService" )
+public class ResourceExtenderService implements IResourceExtenderService
+{
+    /** The Constant BEAN_SERVICE. */
+    // public static final String BEAN_SERVICE = "extend.resourceExtenderService";
+
+    @Inject
+    private IResourceExtenderDAO _extenderDAO;
+
+    @Named( "extend.extendableResourceTypeService" )
+    private IExtendableResourceTypeService _extendableResourceTypeService;
+
+    @Named( "extend.extendableResourceManager" )
+    private IExtendableResourceManager _extendableResourceManager;
+
+    @Inject
+    @LuteceCache( cacheName = "extenderCache", keyType = String.class, valueType = Object.class, enable = true )
+    Lutece107Cache<String, Object> _extenderCache;
```

Comment:
If it's possible to use a type other than Object for the cache value, it's better to go with that solution.